src/material-experimental/mdc-chips/chip-icons.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import {
+  CanDisable,
+  CanDisableCtor,
+  HasTabIndex,
+  HasTabIndexCtor,
+  mixinDisabled,
+  mixinTabIndex,
+} from '@angular/material/core';
+import {FocusableCell} from './grid-focus-key-manager';
+
+
+/**
+ * Directive to add CSS classes to chip leading icon.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-chip-avatar, [matChipAvatar]',
+  host: {
+    'class': 'mat-mdc-chip-avatar mdc-chip__icon mdc-chip__icon--leading',
+    'role': 'img'
+  }
+})
+export class MatChipAvatar {
+  constructor(private _changeDetectorRef: ChangeDetectorRef,
+              private _elementRef: ElementRef) {}
+
+  /** Sets whether the given CSS class should be applied to the leading icon. */
+  setClass(cssClass: string, active: boolean) {
+    const element = this._elementRef.nativeElement;
+    active ? element.addClass(cssClass) : element.removeClass(cssClass);
```

Comment:
In the MDC-based checkbox I did this as an `[ngClass]` binding. (I think the reasoning was that we want to prefer Angular bindings over direct DOM updates whenever possible so that we get a consistent experience in tests where the changes don't get applied until after `fixture.detectChanges()`)

That said, it shouldn't really matter if people are using the component harnesses we plan to build. Those will automatically call `fixture.detectChanges()` on interactions. We should probably be consistent one way or the other though. @jelbourn WDYT, do it like this, or use `[ngClass]`?

Reply:
I was previously trying to do the [ngClass] bindings everywhere, like what was done for checkbox. But it meant that I had to create wrappers in all the template files, since [ngClass] doesn't work on the element host. We eventually moved to DOM updates instead to get rid of the wrappers.

There was a comment here about it:
https://github.com/angular/components/pull/16145#discussion_r294019869

Currently the MatChip component has a similar setMDCClass method to the MatChipAvatar setClass method mentioned here.
https://github.com/angular/components/blob/master/src/material-experimental/mdc-chips/chip.ts#L398

Reply:
Ah that makes sense. I think there's some stuff in Ivy that allows us to bind to `[class]` and have that work just like how `[ngClass]` works now, even for the host element.

We can come back to it once we don't need to support VE anymore, but this is fine for now

Reply:
Yeah, ivy will make this easier, but for now I think this is the best approach